### PR TITLE
[PRMP-1291] Create CRON job to run healthcheck against sandbox daily

### PR DESCRIPTION
**Pull request by oliverbeumkes-nhs**
[PRMP-1291] Create CRON job to run healthcheck against sandbox daily at 4:30AM and then destroy the environment

### CODE REVIEW

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml [hunk 1]
```diff
@@ -1,90 +1,186 @@
...
-#
-#permissions:
-#  pull-requests: write
-#  id-token: write # This is required for requesting the JWT
-#  contents: read # This is required for actions/checkout
+ schedule:
+   - cron: 30 4 * * 1-5
+
+permissions:
+  pull-requests: write
```
Comment (NogaNHS) [inline .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:8]: Does it need this permission?

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml [hunk 2]
```diff
@@ -1,90 +1,186 @@
...
+        uses: actions/checkout@v3
+        with:
+          ref: main
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.AWS_ASSUME_ROLE }}
+          role-skip-session-tagging: true
+          aws-region: ${{ vars.AWS_REGION }}
```
Comment (NogaNHS) [inline .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:29]: Can we use v4 with `mask-aws-account-id` set to true?

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml [hunk 3]
```diff
@@ -1,90 +1,186 @@
...
+
+      - name: View AWS Role
+        run: aws sts get-caller-identity
+
+      # Install the latest version of Terraform CLI and configure the Terraform CLI configuration file with a Terraform Cloud user API token
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v2
```
Comment (NogaNHS) [inline .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:36]: Update this to v3

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml [hunk 4]
```diff
@@ -1,90 +1,186 @@
...
+        with:
+          terraform_version: 1.5.4
```
Comment (NogaNHS) [inline .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:38]: Is it possible not to set a version?

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml (hunk 1 above)
Comment (oliverbeumkes-nhs) [reply .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:8]: I am not sure I would assume no potentially I will try to remove it and check it runs ok

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml (hunk 4 above)
Comment (oliverbeumkes-nhs) [reply .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:38]: I can update to v3 and remove the version but my question would b do we not want the healthcheck to be running the same versions as the other workflows to ensure they are working? Would it not make sense to update them all at the same time fr consistency etc?

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml [hunk 5]
```diff
@@ -1,90 +1,186 @@
...
 jobs:
-  test-job:
+  terraform_process:
     runs-on: ubuntu-latest
+    environment: development
+
+    steps:
+      # Checkout the repository to the GitHub Actions runner
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (abbas-khan10) [inline .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:20]: `actions/checkout@v4`

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml [hunk 6]
```diff
@@ -1,90 +1,186 @@
...
+        with:
+          ref: main
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
```
Comment (abbas-khan10) [inline .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:25]: `aws-actions/configure-aws-credentials@v4`

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml [hunk 7]
```diff
@@ -1,90 +1,186 @@
...
+        with:
+          role-to-assume: ${{ secrets.AWS_ASSUME_ROLE }}
+          role-skip-session-tagging: true
+          aws-region: ${{ vars.AWS_REGION }}
+          mask-aws-account-id: true
+
+      - name: View AWS Role
+        run: aws sts get-caller-identity
+
+      # Install the latest version of Terraform CLI and configure the Terraform CLI configuration file with a Terraform Cloud user API token
```
Comment (abbas-khan10) [inline .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:34]: Would it be ok to remove these comments (and the ones below)? I think the config steps are quite self explanatory for anyone who is able to read the yml

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml [hunk 8]
```diff
@@ -1,90 +1,183 @@
...
+
+      - name: Terraform Init
+        id: init
+        run: terraform init -backend-config=backend.conf
+        working-directory: ./infrastructure
+        shell: bash
+
+      - name: Terraform Set Workspace
+        id: workspace
+        run: terraform workspace select -or-create ndrd
```
Comment (abbas-khan10) [inline .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:47]: Any chance we can put `ndrd` at the top as an `env` var called `workspace`? eg
```
env:  
  WORKSPACE: ndrd
  
- name: Terraform Set Workspace
        id: workspace
        run: terraform workspace select -or-create $WORKSPACE
```

Just so it's clear exactly what and where to set the workspace if this were to ever change in future

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml [hunk 9]
```diff
@@ -1,90 +1,195 @@
...
+  pull-requests: write
+  id-token: write 
+  contents: read
 
 jobs:
-  test-job:
+  env_var:
+    runs-on: ubuntu-latest
+    outputs:
+      workplace: ${{ steps.set-output.outputs.workspace }}
```
Comment (abbas-khan10) [inline .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:19]: *workspace

File: .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml [hunk 10]
```diff
@@ -1,90 +1,195 @@
...
-#  contents: read # This is required for actions/checkout
+
+permissions:
+  pull-requests: write
+  id-token: write 
+  contents: read
 
 jobs:
-  test-job:
+  env_var:
```
Comment (abbas-khan10) [inline .github/workflows/terraform-daily-healthcheck-deploy-and-destroy.yml:16]: Could we rename this to "set_workspace" for clarity?

**Review by abbas-khan10 — APPROVED** [review]

**Pull request merged**